lib/auth.coffee

```diff
@@ -31,6 +31,27 @@ getAuth = (deps, opts) ->
 		else
 			return err
 
+	authSubscribers = []
+
+	exports._onChange = (fn) ->
```

Comment:
I don't really understand the logic behind this, what's the purpose of keeping a list of subscribers? Isn't there only one authenticated client at a time?

Reply:
@sradevski There is indeed only one auth client per SDK instance.
The purpose of this is to provide a way to notify the appropriate caches of the sdk to become invalidated.
This way the auth modules is decoupled from the caches of the various modules that need to be cleared.
I might be missing something, so do you propose something different?

Reply:
@thgreasi Oh now I get it, you are creating a very simple event emitter :). Basically, any model can subscribe to the auth events with a callback, and when an auth change event occurs the subscribers will be notified about it. 

I guess we could go with a proper event handling library, but it might be an overkill for now. I would only suggest to rename `onChange` to `subscribeToEvent` that takes two arguments: event name, and a callback function. I guess for now we can only have a single `eventName` that will be `change` or `*` and the rest will be the same. What do you think?